lib/auth/windows/ldap.go

```diff
@@ -218,7 +275,30 @@ func (c *LDAPClient) ReadWithFilter(dn string, filter string, attrs []string) ([
 	defer c.mu.Unlock()
 
 	res, err := c.client.SearchWithPaging(req, searchPageSize)
-	if err != nil {
+
+	if err == nil {
+		return res.Entries, nil
+	}
+
+	var ldapErr *ldap.Error
+	if !errors.As(err, &ldapErr) {
+		return nil, trace.Wrap(err, "fetching LDAP object %q with filter %q", dn, filter)
+	}
```

Comment:
Maybe it would be cleaner to do this check inside `extractReferrals`, so we'd avoid writing the error string twice.

Reply:
I refactored it a bit to avoid writing the same thing twice